src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/TestUtil.java

```diff
@@ -85,47 +88,44 @@ public static boolean isUtilsClassHasPrivateConstructor(final Class<?> utilClass
+     * @param fieldName the name of the field to retrieve
+     * @return the class' field if found
      */
-    public static Field getClassDeclaredField(Class<?> clss, String fieldName)
-            throws NoSuchFieldException {
-        final Optional<Field> classField = Arrays.stream(clss.getDeclaredFields())
-                .filter(field -> fieldName.equals(field.getName())).findFirst();
-        final Field resultField;
-        if (classField.isPresent()) {
-            resultField = classField.get();
-        }
-        else {
-            resultField = clss.getSuperclass().getDeclaredField(fieldName);
-        }
-        resultField.setAccessible(true);
-        return resultField;
+    public static Field getClassDeclaredField(Class<?> clss, String fieldName) {
+        return Stream.<Class<?>>iterate(clss, Class::getSuperclass)
+            .flatMap(cls -> Arrays.stream(cls.getDeclaredFields()))
+            .filter(field -> fieldName.equals(field.getName()))
+            .peek(field -> field.setAccessible(true))
+            .findFirst()
+            .orElseThrow(() -> {
+                return new IllegalStateException(String.format(Locale.ROOT,
+                        "Field '%s' not found in '%s'", fieldName, clss.getCanonicalName()));
+            });
     }
 
     /**
      * Retrieves the specified method by it's name in the class or it's direct super.
      *
-     * @param clss The class to retrieve the field for.
-     * @param methodName The name of the method to retrieve.
-     * @return The class' field.
-     * @throws NoSuchMethodException if the requested method cannot be found in the class.
+     * @param clss the class to retrieve the method for
+     * @param methodName the name of the method to retrieve
+     * @param parameters the expected number of parameters
+     * @return the class' method
      */
-    public static Method getClassDeclaredMethod(Class<?> clss, String methodName)
-            throws NoSuchMethodException {
-        final Optional<Method> classMethod = Arrays.stream(clss.getDeclaredMethods())
-                .filter(method -> methodName.equals(method.getName())).findFirst();
-        final Method resultMethod;
-        if (classMethod.isPresent()) {
-            resultMethod = classMethod.get();
-        }
-        else {
-            resultMethod = clss.getSuperclass().getDeclaredMethod(methodName);
-        }
-        resultMethod.setAccessible(true);
-        return resultMethod;
+    public static Method getClassDeclaredMethod(Class<?> clss, String methodName, int parameters) {
+        return Stream.<Class<?>>iterate(clss, Class::getSuperclass)
+            .flatMap(cls -> Arrays.stream(cls.getDeclaredMethods()))
+            .filter(method -> {
+                return methodName.equals(method.getName())
+                    && parameters == method.getParameterCount();
+            })
+            .peek(method -> method.setAccessible(true))
+            .findFirst()
```

Comment:
Can we do findfirs fist and then peek ?

Reply:
Done. `Optional` has no method `peek` so it is done with `map` method.